### ISSUE TITLE
lua-SDL2 syntax list?

### ISSUE THREAD

**Issue by lewislepton** [Tangent128/luasdl2#21]
rather than carry on the yosemite build one, thought id ask here.

but where could i get a list of the lua-SDL2 syntax? with everything. ill work on a cheap auto-complete for it to make it easier coding on say, sublime, or whatever else. just dont know how to get the syntax list. just so when future people use this, there is an option.

thanks

**Comment by Tangent128** [comment]
You mean a list of the symbols used in the API?

Technically, everything is in the [wiki](https://github.com/Tangent128/luasdl2/wiki), but that can be a bit hard to parse.

Here is a list of all the keys in the SDL table & all metatables: [gist](https://gist.github.com/Tangent128/ffc11d02722fe7d2f9c3)

You probably want to remove the first column with cut/awk/sed/whatever, but I left it on in case the context is useful to you.

**Comment by lewislepton** [comment]
sweet. thanks man.
aye, just a way really to print out functions, classes ec etc that make using lua-sdl2. true it all in the API, but a list is good, because then it can be looked through then edited once printed

thanks

**Comment by lewislepton** [comment]
really its all the stuff that has ```SDL.whatever``` and things like ```KeyPressed.whatever```. like if you were on any IDE and it popped up with a list once you starte typing

ill give this a look

**Comment by lewislepton** [comment]
alright man.

made a auto-complete for sublime. could be used with anything really, its just a JSON table. but here if you are interested
https://github.com/lewislepton/Lua-SDL2AutoComplete

**Comment by lewislepton** [comment]
also put a link for this repo as well with it ;)

**Comment by Tangent128** [comment]
Added a link on the wiki.
